src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/entity/PlanAgreementNoteEntity.kt

```diff
@@ -37,7 +37,7 @@ class PlanAgreementNoteEntity(
   var planVersion: PlanVersionEntity?,
 
   @Column(name = "optional_note")
-  var optionalNote: String,
+  var optionalNote: String? = null,
```

Comment:
Not sure about this, whether we should just pass an empty string instead or allow it to be truly optional.

Reply:
iirc the way it was working in practice is that we were actually getting an empty string through the API so it didn't need to be nullable, but I think a null would actually be better. Do we need to update some other tests to make sure the value of this field really is null if an optional note is not provided?

Reply:
I've reverted this change for now, with the view to take a look at it once this has been done. It didn't actually offer us any benefit for this because all of the other structures will need changing too for it to be possible for us to omit this field altogether